src/Application/ApplicationFileProcessor.php

```diff
@@ -42,23 +60,38 @@ public function __construct(
      * @param File[] $files
      * @return array{system_errors: SystemError[], file_diffs: FileDiff[]}
      */
-    public function run(array $files, Configuration $configuration): array
+    public function run(array $files, Configuration $configuration, InputInterface $input): array
     {
+        // no files found
+        $filesCount = count($files);
+        if ($filesCount === 0) {
```

Comment:
can use `$files === []` for performance improvement:

```suggestion
        if ($files === []) {
```